### PR TITLE
Fix bad page on search

### PR DESCRIPTION
**Pull request by arthurnn**
This fixes pagination issues we see, when sending a `page=0` parameter or `page=foo`

Would like to merge #851 before this, as this includes that extra commit from there.

### CODE REVIEW

**Comment by dwradcliffe** [comment]
Looks good after rebase.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -65,4 +68,9 @@ def find_rubygem_by_name
              :status => :not_found
     end
   end
+
+  def set_page
+    @page = [1, params[:page].to_i].max
+    true
```
Comment (simi) [inline app/controllers/application_controller.rb]: Is the `true` really needed here?

Comment (qrush) [inline app/controllers/application_controller.rb]: Shouldn't be. :gun:

**Comment by arthurnn** [comment]
Fixed the reviewed points. last look? @simi / @dwradcliffe

**Comment by simi** [comment]
Go @arthurnn!

I was thinking about lazy loaded `current_page`, but I can try to improve this and open discussion later.

**Comment by dwradcliffe** [comment]
:ship:

**Pull request merged**